windows/ReactNativeBlobUtil/ReactNativeBlobUtil.cpp

```diff
@@ -108,1406 +110,1707 @@ void TaskCancellationManager::Cancel(TaskId taskId) noexcept
+                folderOp.get().DeleteAsync().get();
+            }
+            else
+            {
+                winrt::hstring directoryPath, fileName;
+                splitPath(path, directoryPath, fileName);
+                auto folder = winrt::Windows::Storage::StorageFolder::GetFolderFromPathAsync(directoryPath).get();
+                auto item = folder.GetItemAsync(fileName).get();
+                item.DeleteAsync().get();
+            }
+
+            ::React::JSValueArray result;
+            result.push_back(::React::JSValue(true));
+            callback(std::move(result));
+        }
+        catch (const winrt::hresult_error& ex)
+        {
+            ::React::JSValueArray errorResult;
+            errorResult.push_back(::React::JSValue(false));
+            errorResult.push_back(::React::JSValue(winrt::to_string(ex.message())));
+            callback(std::move(errorResult));
+        }
+    });
+}
+
+void ReactNativeBlobUtil::removeSession(
+    ::React::JSValueArray&& paths,
+    std::function<void(::React::JSValueArray const&)> const& callback) noexcept
 {
-	callback(winrt::to_string(ex.message().c_str()));
+    winrt::Windows::System::Threading::ThreadPool::RunAsync([paths = std::move(paths), callback](auto&&)
+    {
+        ::React::JSValueArray resultArray;
+        try
+        {
+            for (const auto& path : paths)
+            {
+                std::filesystem::path toDelete{ path.AsString() };
+                toDelete.make_preferred();
+                auto fileOp = winrt::Windows::Storage::StorageFile::GetFileFromPathAsync(winrt::to_hstring(toDelete.c_str()));
+                auto file = fileOp.get();
+                file.DeleteAsync().get();
+            }
+            // Success: return empty array
+            callback(resultArray);
+        }
+        catch (const winrt::hresult_error& ex)
+        {
+            resultArray.push_back("EUNSPECIFIED");
+            resultArray.push_back(winrt::to_string(ex.message()));
+            callback(resultArray);
+        }
+    });
 }
 
-void ReactNativeBlobUtil::writeArrayChunk(
-	std::string streamId,
-	winrt::Microsoft::ReactNative::JSValueArray dataArray,
-	std::function<void(std::string)> callback) noexcept
-	try
+winrt::fire_and_forget ReactNativeBlobUtil::ls(
+    std::string path,
+    ::React::ReactPromise<::React::JSValueArray> promise) noexcept
 {
-	auto stream{ m_streamMap.find(streamId)->second };
-	std::vector<uint8_t> data;
-	data.reserve(dataArray.size());
-	for (auto& var : dataArray)
-	{
-		data.push_back(var.AsUInt8());
-	}
-	Streams::IBuffer buffer{ CryptographicBuffer::CreateFromByteArray(data) };
-
-	stream.streamInstance.WriteAsync(buffer).get(); // Calls it synchronously
-	callback("");
+    try
+    {
+        winrt::hstring directoryPath, fileName;
+        splitPath(path, directoryPath, fileName);
+
+        Windows::Storage::StorageFolder::GetFolderFromPathAsync(directoryPath).Completed(
```

Comment:
Now that this function is fire_and_forget, you can switch from using completed callbacks to co_await